sql/rpl_rli.h

```diff
@@ -621,10 +627,14 @@ struct inuse_relaylog {
   */
   rpl_gtid *relay_log_state;
   uint32 relay_log_state_count;
-  /* Number of events in this relay log queued for worker threads. */
-  Atomic_counter<int64> queued_count;
-  /* Number of events completed by worker threads. */
-  Atomic_counter<int64> dequeued_count;
+  /** Number of events in this relay log queued for worker threads.
+    @pre Hold `rli->data_lock` while accessing
+  */
+  int64_t queued_count;
+  /** Number of events completed by worker threads.
+    @pre Hold `rli->data_lock` while accessing
+  */
+  int64_t dequeued_count;
```

Comment:
Now that every access has a reason to hold `rli->data_lock`, this pair no longer needs to be atomic.

Reply:
Obsolete: The new solution has reverted the reasons that are new to this change because [they were a performance toll](https://github.com/MariaDB/server/pull/4103#issuecomment-2998129259).